File: python/1.py
import collections
from typing import List

class Solution(object):
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_map = collections.defaultdict(int)
        for i, n in enumerate(nums):
            num_map[n] = i

        num1: int
        num2: int
        for i, n in enumerate(nums):
            # if num_map[target - n] and num_map[target - n] != i:
            if target - n in num_map and num_map[target - n] != i:
                num1 = i
                num2 = num_map[target - n]
                break

        return [num1, num2]


if __name__ == "__main__":
    solution = Solution()

    # num_arr = [2, 7, 11, 15]
    # tar = 9

    num_arr = [2, 5, 5, 11]
    tar = 10
    print(solution.twoSum(num_arr, tar))

File: cpp/1.cpp

#include <vector>
#include <iostream>
#include <map>

using namespace std;

vector<int> twoSum(vector<int> &nums, int target) {
    map<int, int> numMap;
    vector<int> output;

    int n = nums.size();
    for (int i = 0; i < n; i++) {
        numMap[nums[i]] = i; 
    }

    for (int i = 0; i < n; i++) {
        if (numMap.find(target - nums[i]) != numMap.end() && 
            numMap[target - nums[i]] != i) {
                output.push_back(i);
                output.push_back(numMap[target - nums[i]]);
                break;
            }
    }
    return output;

}

int main() {

    vector<int> input = {2, 7, 11, 15};
    int target = 22;

    vector<int> output = twoSum(input, target);

    for (auto i : output) {
        cout << i << endl;
    }
    

    return 0;
}